File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution 
{
    public ListNode removeNthFromEnd(ListNode head, int n) 
    {

     ListNode dummy = new ListNode(0);
     dummy.next = head;
     ListNode first = dummy;
     ListNode second = dummy;


     while(first!=null && n>=0)
     {
        first = first.next;
        n-=1;
     }

     while(first!=null)
     {
        second = second.next;
        first = first.next;
     }

     second.next = second.next.next;

     return dummy.next;

    }
}

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        slow_ptr: ListNode = head
        fast_ptr: ListNode = head
        
        while slow_ptr is not None and fast_ptr is not None and fast_ptr.next is not None:
            slow_ptr = slow_ptr.next
            fast_ptr = fast_ptr.next.next

            if (slow_ptr is not None and fast_ptr is not None) and (slow_ptr.val == fast_ptr.val):
                return True
        return False
